File: ios/TestKit/TestKit/Sources/Mocks/Features/Story/Model/StoryManagerMock.swift
import NetworkKit
import StoryKit
import XCTest

public struct StoryManagerMock {
    public var result: Result<Story, Error>

    public init(result: Result<Story, Error> = .failure(NetworkError.invalidResponse)) {
        self.result = result
    }
}

extension StoryManagerMock: StoryManagerProtocol {
    public func story(_ observer: @escaping Observer<Story>) {
        observer(result)
    }
}

File: android/SwiftKotlination/features/story/src/test/kotlin/fr/jhandguy/story/model/mocks/StoryManagerMock.kt
package fr.jhandguy.story.model.mocks

import fr.jhandguy.network.model.observer.Observer
import fr.jhandguy.network.model.observer.Result
import fr.jhandguy.story.model.Story
import fr.jhandguy.story.model.StoryManagerInterface

class StoryManagerMock(val result: Result<Story>) : StoryManagerInterface {
    override suspend fun story(observer: Observer<Story>) = observer(result)
}
